kaitaistruct.lua

```diff
@@ -221,10 +230,6 @@ end
 -- Deprecated, use read_bits_int_be() instead.
 --
 function KaitaiStream:read_bits_int(n)
-    return self:read_bits_int_be(n)
-end
-
-function KaitaiStream:read_bits_int_le(n)
```

Comment:
Yes, I took a lot of code over from luajit_runtime repo, but there must have been reason for me to do it this way, it's hard to remember after few months since i've done it.
Strangely, this did not mess up any tests in place

Also no, i'm not going to rewrite the functions using `bit` module from scratch now, but we can surely add that task to backlog of other issues, this runtime has.

Reply:
> there must have been reason for me to do it this way 

I really doubt that. It's obviously wrong, please read my above comment. 

> Also no, i'm not going to rewrite the functions using `bit` module from scratch now, but we can surely add that task to backlog of other issues, this runtime has.

Sorry, maybe I wrote it ambiguously, I meant copying the [`read_bits_int`](https://github.com/kaitai-io/kaitai_struct_lua_runtime/blob/648ed1cf5ca1c9dd8bdf55787b1646b5cfa18def/kaitaistruct.lua#L218-L225) and [`read_bits_int_le`](https://github.com/kaitai-io/kaitai_struct_lua_runtime/blob/648ed1cf5ca1c9dd8bdf55787b1646b5cfa18def/kaitaistruct.lua#L227-L251) metho

Reply:
Yeah, ok, that's something i can do for sure, no hard feelings :+1:

Reply:
> Strangely, this did not mess up any tests in place

It certainly would now. How would possibly pass the test [`bits_simple_le`](https://github.com/kaitai-io/ci_targets/blob/6aca1b7283577518a378a52f84387d86d7f646c6/compiled/lua/bits_simple_le.lua#L18) when you completely removed the `read_bit_ints_le` method that it depends on?

Reply:
> Yeah, that's what happens, if my contribution from 13th is compared with updated tests/expectations from 23rd August

That's not true. _Expectations_ on this matter have been last updated with these commits:

- [kaitai-io/kaitai_struct_lua_runtime@`fc9556d`](https://github.com/kaitai-io/kaitai_struct_lua_runtime/commit/fc9556d32c5d6bdf03177230afcd244961415ffd) implementing the method `read_bits_int_le` (from April 28, 2020 0:27)
- [kaitai-io/kaitai_struct_compiler@`746e34e`](https://github.com/kaitai-io/kaitai_struct_compiler/commit/746e34e4ac672c64a50727acc560c665e5cf2083#diff-18b3799f9c0ec3e3b1343e53ac35edcf) making the compiler output `read_bits_int_{be,le}()` calls in the generated Lua parsers (from April 29, 2020 21:17)
- [kaitai-io/ci_targets@`27d5f3b`](https://github.com/kaitai-io/ci_targets/commit/27d5f3b1f83f443f588ffd4bd0e6eae5d6be6170#diff-fcadf51613a227e6294f58e7100e2bb9) with the CI rebuild of `bits_simple_le.lua` parser code based on test KSY spec [`bits_simple_le.ksy`](https://github.com/kaitai-io/kaitai_struct_tests/blob/3cb98ecd6d7793ce412d53b210bc6f4b595ed544/formats/bits_simple_le.ksy) using KSC having the change above - it shows that `read_bits_int_le` method is already in use (commit from April 29, 2020 21:40)

If this PR _would_ be based on a earlier commit than https://github.com/kaitai-io/kaitai_struct_lua_runtime/commit/fc9556d32c5d6bdf03177230afcd244961415ffd (**imaginary** situation), Git would not allow to merge this PR automatically, and merge conflicts would occur instead.

Reply:
Sure, you're right, hopefully you feel satisfied

Reply:
Fixed with 6f1e309cf1fb48efe55568becd02b9ffb6306ed9